### PR TITLE
Adding ability to stop restart loop on fatal initialization failures. Updating language check to throw when in mixed mode

### PR DESCRIPTION
**Pull request by fabiocav**
Working on tests. Just getting this up for feedback.

This changes the behavior of the single language check so that it throws and places the host in an error state until the problem is resolved. This prevents the startup from running indefinitely when in a permanently broken state.

### CODE REVIEW

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 1]
```diff
@@ -660,10 +660,9 @@ internal Collection<FunctionDescriptor> GetFunctionDescriptors(IEnumerable<Funct
             Collection<FunctionDescriptor> functionDescriptors = new Collection<FunctionDescriptor>();
             var httpFunctions = new Dictionary<string, HttpTriggerAttribute>();
 
-            if (!Utility.IsSingleLanguage(functions, _currentRuntimelanguage))
+            if (!_scriptHostEnvironment.IsDevelopment() && !Utility.IsSingleLanguage(functions, _currentRuntimelanguage))
             {
-                _logger.LogError($"Found functions with more than one language. Select a language for your function app by specifying {LanguageWorkerConstants.FunctionWorkerRuntimeSettingName} AppSetting");
-                return functionDescriptors;
+                throw new HostInitializationException($"Found functions with more than one language. Select a language for your function app by specifying {LanguageWorkerConstants.FunctionWorkerRuntimeSettingName} AppSetting");
```
Comment (pragnagopa) [inline src/WebJobs.Script/Host/ScriptHost.cs]: LGTM. Assuming you will be adding a test that verifies throwing HostInitializationException

File: src/WebJobs.Script.WebHost/DefaultScriptHostBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+using System;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public sealed class DefaultScriptHostBuilder : IScriptHostBuilder
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/DefaultScriptHostBuilder.cs]: Introducing the implementation and interface to make testing a little easier for the hosted service. This was pretty cumbersome to setup.

File: src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs [hunk 3]
```diff
@@ -74,29 +78,32 @@ public async Task StartAsync(CancellationToken cancellationToken)
...
                 {
                     LastError = null;
-                }
 
-                if (!isOffline && !skipHostJsonConfiguration)
-                {
-                    State = ScriptHostState.Running;
+                    if (!isOffline)
+                    {
+                        State = ScriptHostState.Running;
```
Comment (brettsam) [inline src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs]: If we've skipped host startup due to an error, do we still want it to say Running?

**Comment by fabiocav** [comment]
@mathewc @brettsam, let me know if you have any thoughts on this.

File: src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs (hunk 3 above)
Comment (fabiocav) [reply src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs]: This is only hit if not in an error state. See the `if` above. I added a test that validates that `State` will be error and the exception will be set.

**Pull request merged**